src/Blumchen/DependencyInjection/Worker.cs

```diff
@@ -0,0 +1,48 @@
+using System.Collections.Concurrent;
+using Blumchen.Subscriptions;
+using Blumchen.Subscriptions.Replication;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+
+namespace Blumchen.DependencyInjection;
+
+public class Worker<T>(
+    WorkerOptions options,
+    ILogger<Worker<T>> logger): BackgroundService where T : class, IMessageHandler
+{
+    private string WorkerName { get; } = $"{nameof(Worker<T>)}<{typeof(T).Name}>";
+    private static readonly ConcurrentDictionary<string, Action<ILogger, string, object[]>> LoggingActions = new(StringComparer.OrdinalIgnoreCase);
+    private static void Notify(ILogger logger, LogLevel level, string template, params object[] parameters)
```

Comment:
**QUESTION:** Why do we have such a wrapper for regular logging handling? What's the additional benefit of just using a logger?

Reply:
We don’t have to always verify the enabled logging level.
I’ll check the source generated version for comparison